src/segment-loader.js

```diff
@@ -848,7 +849,7 @@ export default class SegmentLoader extends videojs.EventTarget {
       return false;
     }
 
-    if (this.loaderType === 'main' &&
+    if (this.loaderType_ === 'main' &&
```

Comment:
btw, this is the main fix for this PR. Everything else just helps with debugging.